src/engine/bpmcontrol.cpp

```diff
@@ -849,6 +849,11 @@ void BpmControl::collectFeatures(GroupFeatureState* pGroupFeatures) const {
 
         pGroupFeatures->has_beat_fraction = true;
         pGroupFeatures->beat_fraction = dThisBeatFraction;
+
+        pGroupFeatures->has_bpm = true;
+        pGroupFeatures->bpm = getBpm();
+
+        pGroupFeatures->isPlaying = m_pPlayButton->get();
```

Comment:
"Is plaing"  I specific somehow for deck inputs. Can we make this his more generic?
Like setting has beat_fraction to false if it is not playing? 
I think that is what you actually need.

Reply:
What happen if you scratch a paused deck, do we expect an synced effect to work?

Reply:
> setting has beat_fraction to false if it is not playing?

Setting has_beat_fraction to false when the track is not playing will work with the current native effects (metronome, tremolo, auto pan). But I think we should keep it as it is so that the beat_fraction can be used by effects in a paused state also. What do you think?

Reply:
> What happen if you scratch a paused deck, do we expect an synced effect to work?

Scratching a paused deck won't change the phase of the metronome. The phase is being set only when the track is playing. This allows us to seek without disturbing the metronome.

Reply:
But isn't it a feature to hear the metronome when cueing the track.